test/Williamson2_ShallowWaterExplicit.jl

```diff
@@ -0,0 +1,114 @@
+module Williamson2_ShallowWaterExplicit
+
+using FillArrays
+using Test
+using WriteVTK
+using Gridap
+using GridapGeosciences
+
+# Solves the steady state Williamson2 test case for the shallow water equations on a sphere
+# of physical radius 6371220m. Involves a modified coriolis term that exactly balances
+# the potential gradient term to achieve a steady state
+# reference:
+# D. L. Williamson, J. B. Drake, J. J.HackRüdiger Jakob, P. N.Swarztrauber, (1992)
+# J Comp. Phys. 102 211-224
+
+# Constants of the Williamson2 test case
+const α  = π/4.0              # deviation of the coriolis term from zonal forcing
+const U₀ = 38.61068276698372  # velocity scale
+const H₀ = 2998.1154702758267 # mean fluid depth
+
+# Modified coriolis term
+function f₀(xyz)
+   θϕr   = xyz2θϕr(xyz)
+   θ,ϕ,r = θϕr
+   2.0*Ωₑ*( -cos(θ)*cos(ϕ)*sin(α) + sin(ϕ)*cos(α) )
+end
+
+# Initial velocity
+function u₀(xyz)
+  θϕr   = xyz2θϕr(xyz)
+  θ,ϕ,r = θϕr
+  u     = U₀*(cos(ϕ)*cos(α) + cos(θ)*sin(ϕ)*sin(α))
+  v     = -U₀*sin(θ)*sin(α)
+  spherical_to_cartesian_matrix(θϕr)⋅VectorValue(u,v,0)
+end
+
+# Initial fluid depth
+function h₀(xyz)
+  θϕr   = xyz2θϕr(xyz)
+  θ,ϕ,r = θϕr
+  h  = -cos(θ)*cos(ϕ)*sin(α) + sin(ϕ)*cos(α)
+  H₀ - (rₑ*Ωₑ*U₀ + 0.5*U₀*U₀)*h*h/g
+end
+
+l2_err_u = [0.011504453807392859, 0.003188984305055811, 0.0008192298898147198]
+l2_err_h = [0.005636335001937436, 0.0014571807037802682, 0.0003681933640549439]
+
+function forward_step(i, n)
+  order = 1
+  degree = 4
+
+  model = CubedSphereDiscreteModel(n, order+1, radius=rₑ)
```

Comment:
Remove `order+1` from this call, so that we use the analytical mapping and the AD Jacobian, i.e., we do not approximate the geometry of the sphere.

Reply:
Currently still using the isoparametric Jacobian in order to reduce under integration of the metric terms. However I take your point that there will still be under-integration of the Gammian even for the isoparametric Jacobian, so I will change over to the AD Jacobian...